### PR TITLE
Researcher Database Feature

### LINKED ISSUE #735: Funcionalidade de Banco de pesquisadores
## Feature Description

  * Fazer uma lista dos pesquisadores em suas áreas de atuação
  * Somente pesquisadores que aprovarem o seu nome no banco de pesquisadores (Checkbox no Perfil do pesquisador)
  * Ao entrar na tela apresentar um campo de busca
  * A busca será por nome, organização, área de conhecimento, palavra-chave de suas tecnologias cadastradas
  * Os campos que irão apresentar na lista de resultados será:
     - Nome completo
     - Instituição
     - Área do conhecimento
     - Palavras-chave das tecnologias que o pesquisador é o responsável
     - Link para a plataforma Lattes (abrir em uma nova janela)
     - Ícone para entrar em contato com o pesquisador
       - Abre um modal com a opção de enviar mensagem para o pesquisador (título e mensagem) 
       - Somente usuários cadastrados e logados podem enviar mensagem  

---------------

## Acceptance criteria

* 

## Implementation Brief

*

### PR DESCRIPTION
**Pull request by alexandreadames**
## Summary

Resolves #735

### Rotas alteradas

| Description | Method | Route | Body/Query | Auth |
| --- | --- | --- | --- | --- |
Creates a new User|POST|users | New fields: researcher (boolean)*, areas (knowledge area list)| YES: CREATE_USERS | 
Updates a User|PUT|users/:id | New fields: researcher (boolean)*, areas (knowledge area list)| YES: UPDATE_USER or UPDATE_USERS | 

*Identifica se o usuário optou por incluir seu registro no banco de pesquisadores.

### Rota Nova

| Description | Method | Route | Body/Query | Auth |
| --- | --- | --- | --- | --- |
Searches in Researcher database|GET|researchers | Query Params for filtering: name (first name or last name), institution (name), keyword (string), area (knowledge area list) | No | 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code passes the linting.
- [x] My code has proper inline documentation.
- [x] I have manually tested this PR.

### CODE REVIEW

File: packages/api/app/Controllers/Http/ResearcherController.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+			.where({ researcher: true })
+			.with('institution')
+			.with('areas')
+			.with('technologies', (builder) => {
+				builder.where('status', 'published');
+				builder.wherePivot('role', 'OWNER');
+				builder.with('terms', (builder2) => {
+					builder2.where('taxonomy_id', keywordTaxonomy.id);
+				});
+			})
```
Comment (mateus4k) [inline packages/api/app/Controllers/Http/ResearcherController.js:18]:
```suggestion
			.with('technologies', (builder) => {
				builder.where('status', technologyStatuses.PUBLISHED);
					.wherePivot('role', roles.OWNER);
					.with('terms', (query) => {
						query.where('taxonomy_id', keywordTaxonomy.id);
					});
			})
```

File: packages/api/app/Controllers/Http/ResearcherController.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+			.withParams(request, { skipRelationships: ['technologies'] });
+		const researchers = users.rows.map((user) => ({
+			full_name: user.toJSON().full_name,
+			institution: user.toJSON().institution.name,
+			areas: user.toJSON().areas,
+			keywords: user
+				.toJSON()
+				.technologies.map((technology) => technology.terms)
+				.flat(1),
+			lattes_link: `http://lattes.cnpq.br/${user.toJSON().lattes_id}`,
```
Comment (mateus4k) [inline packages/api/app/Controllers/Http/ResearcherController.js:29]: O que acha de extrair essa url base do lattes para um arquivo de config? 🤔

File: packages/api/app/Controllers/Http/ResearcherController.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+				builder.with('terms', (builder2) => {
+					builder2.where('taxonomy_id', keywordTaxonomy.id);
+				});
+			})
+			.withResearcherFilters(filters)
+			.withParams(request, { skipRelationships: ['technologies'] });
+		const researchers = users.rows.map((user) => ({
+			full_name: user.toJSON().full_name,
+			institution: user.toJSON().institution.name,
+			areas: user.toJSON().areas,
```
Comment (mateus4k) [inline packages/api/app/Controllers/Http/ResearcherController.js:24]: Acho que seria melhor dar um `toJSON()` antes do map, assim, acessaria todas as variáveis normalmente

```suggestion
		const researchers = users.toJSON().map((user) => ({
			full_name: user.full_name,
			institution: user.institution.name,
			areas: user.areas,
```

**Review by mateus4k — DISMISSED** [review]
Aprovado! Apenas algumas sugestões caso ache interessante 👍

**Review by mateus4k — APPROVED** [review]
💯

**Review by NaylsonFerreira — APPROVED** [review]

**Pull request merged**